addons/resource/tests/test_resource.py

```diff
@@ -6,351 +6,290 @@
 import babel.dates
 from dateutil.relativedelta import relativedelta
 
-from openerp.addons.resource.tests.common import TestResourceCommon
+from odoo.fields import Datetime
+from odoo.tools import float_compare
+from odoo.addons.resource.tests.common import TestResourceCommon
 
 from openerp.tests import TransactionCase
 
 
 class TestResource(TestResourceCommon):
 
     def test_00_intervals(self):
-        cr, uid = self.cr, self.uid
         intervals = [
             (
-                datetime.strptime('2013-02-04 09:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-04 11:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-04 09:00:00'),
+                Datetime.from_string('2013-02-04 11:00:00')
             ), (
-                datetime.strptime('2013-02-04 08:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-04 12:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-04 08:00:00'),
+                Datetime.from_string('2013-02-04 12:00:00')
             ), (
-                datetime.strptime('2013-02-04 11:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-04 14:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-04 11:00:00'),
+                Datetime.from_string('2013-02-04 14:00:00')
             ), (
-                datetime.strptime('2013-02-04 17:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-04 21:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-04 17:00:00'),
+                Datetime.from_string('2013-02-04 21:00:00')
             ), (
-                datetime.strptime('2013-02-03 08:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-03 10:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-03 08:00:00'),
+                Datetime.from_string('2013-02-03 10:00:00')
             ), (
-                datetime.strptime('2013-02-04 18:00:00', '%Y-%m-%d %H:%M:%S'),
-                datetime.strptime('2013-02-04 19:00:00', '%Y-%m-%d %H:%M:%S')
+                Datetime.from_string('2013-02-04 18:00:00'),
+                Datetime.from_string('2013-02-04 19:00:00')
             )
         ]
 
         # Test: interval cleaning
-        cleaned_intervals = self.resource_calendar.interval_clean(intervals)
+        cleaned_intervals = self.ResourceCalendar._model.interval_clean(intervals)
```

Comment:
ResourceCalendar is already self.env['resource.calendar']. I don't see why you access to the _model